tests/xharness/Utilities/StringUtils.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Xharness.Utilities
+{
+	class StringUtils
+	{
+		static readonly char shellQuoteChar;
+		static readonly char [] mustQuoteCharacters = new char [] { ' ', '\'', ',', '$', '\\' };
+		static readonly char [] mustQuoteCharactersProcess = { ' ', '\\', '"', '\'' };
+
+		static StringUtils ()
+		{
+			PlatformID pid = Environment.OSVersion.Platform;
+			if ((int) pid != 128 && pid != PlatformID.Unix && pid != PlatformID.MacOSX)
+				shellQuoteChar = '"'; // Windows
+			else
+				shellQuoteChar = '\''; // !Windows
+		}
+
+		public static string FormatArguments (params string [] arguments)
+		{
+			return FormatArguments ((IList<string>) arguments);
+		}
+
+		public static string FormatArguments (IList<string> arguments)
+		{
+			return string.Join (" ", QuoteForProcess (arguments));
+		}
+
+		private static string [] QuoteForProcess (params string [] array)
```

Comment:
The Mono Coding Guidelines says to not use `private` (because it's redundant).

```suggestion
	static string [] QuoteForProcess (params string [] array)
```